exp/runtime/hooks/api/v1alpha1/lifecyclehooks_types.go

```diff
@@ -136,7 +181,23 @@ type BeforeClusterDeleteRequest struct {
 	Cluster clusterv1.Cluster `json:"cluster"`
 }
 
-func BeforeClusterDelete(*BeforeClusterDeleteRequest, *BlockingResponse) {}
+// BeforeClusterDeleteResponse is the response of BeforeClusterDelete hook.
+// +kubebuilder:object:root=true
+type BeforeClusterDeleteResponse struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Status of the call. One of "Success" or "Failure".
+	Status ResponseStatus `json:"status"`
+
+	// RetryAfterSeconds when set to a non-zero signifies that the hook
+	// needs to be retried at a future time.
+	RetryAfterSeconds int `json:"retryAfterSeconds"`
+
+	// A human-readable description of the status of the call.
+	Message string `json:"message"`
+}
+
+func BeforeClusterDelete(*BeforeClusterDeleteRequest, *BeforeClusterDeleteResponse) {}
```

Comment:
There's a few typos in the init function - I can't comment directly on them :) but a few examples:

crated -> created
add 'blocking' to the description for BeforeClusterUpgrade ('This blocking hook')
double space in description for BeforeClusterUpgrade